### PR TITLE
support for ne04j 5.x

### PR DESCRIPTION
**Pull request by PranjaliParnerkar**
@paltusplintus Please review.
@Vidhisri56 @KWLgsk FYI.

This PR is for this [PBI](https://dev.azure.com/DevOps-RD/Clinical%20Linked%20Data%20Experiments/_sprints/taskboard/Clinical%20Linked%20Data%20Experiments%20Team/Clinical%20Linked%20Data%20Experiments/Sprint%2012.1?workitem=1198654).

This adds support for neo4j5.x.

### CODE REVIEW

File: derivation_method/derivation_method.py [hunk 1]
```diff
@@ -892,7 +907,8 @@ def _predict_links(self) -> tuple:
         logger.info(f'\tPredicting links actions for {self.name}')
 
         filepath = os.path.dirname(__file__)
-        with open(os.path.join(filepath, 'predict_links.cql')) as cypherfile:
+        filename = 'predict_links.cql' if re.search(r'^[01234]\.', self.interface.get_dbms_details()[0]['version']) else 'predict_links_neo4j5.x.cql'
```
Comment (paltusplintus) [inline derivation_method/derivation_method.py:910]: IS NULL works in 4.x as well, so we probably can just have 1 file

File: derivation_method/super_method.py [hunk 2]
```diff
@@ -930,7 +930,7 @@ def retrieve_json(self):
                     denominator: is_denominator,
                     required: is_required,
                     `all_ct`: should_all_ct
-                }
+                }, x_) YIELD rel
```
Comment (paltusplintus) [inline derivation_method/super_method.py:933]: does this work in both 4.x and 5.x?

File: derivation_method/derivation_method.py [hunk 3]
```diff
@@ -222,11 +230,18 @@ def check_method_actions_order(self):
...
+        if re.search(r'^[01234]\.', self.interface.get_dbms_details()[0]['version']):
+            q = """
+            MATCH (m:Method{id:$method_id})
+            WHERE NOT EXISTS (m.parent_id)
+            RETURN id(m)
+            """
+        else:
+            q = """
+            MATCH (m:Method{id:$method_id})
+            WHERE m.parent_id IS NULL
```
Comment (paltusplintus) [inline derivation_method/derivation_method.py:242]: same here

**Review by paltusplintus — CHANGES_REQUESTED** [review]
@PranjaliParnerkar Thanks! Please see some comments below

File: derivation_method/super_method.py (hunk 2 above)
Comment (PranjaliParnerkar) [reply derivation_method/super_method.py:933]: Yes, I don't know how was this missed but this test case was failing for 4.x earlier.

**Comment by PranjaliParnerkar** [comment]
@paltusplintus Please review!

The test case **test_derivation_method.py > test_build_derivation_method_json** is still not working due to some previous issue which is not version specific. Can we create a bugfix for this in next sprint as the debugging would take time. Thanks!

**Comment by paltusplintus** [comment]
> @paltusplintus Please review!
> 
> The test case **test_derivation_method.py > test_build_derivation_method_json** is still not working due to some previous issue which is not version specific. Can we create a bugfix for this in next sprint as the debugging would take time. Thanks!

ok. thanks is failing is test_subject_level_link.json - it requires class parameter with a Term{rdfs:label: 'Age'} - but I could not fix it quickly either

**Comment by paltusplintus** [comment]
> > @paltusplintus Please review!
> > The test case **test_derivation_method.py > test_build_derivation_method_json** is still not working due to some previous issue which is not version specific. Can we create a bugfix for this in next sprint as the debugging would take time. Thanks!
> 
> ok. thanks is failing is test_subject_level_link.json - it requires class parameter with a Term{rdfs🏷️ 'Age'} - but I could not fix it quickly either

The test  
tests/derivation_method/test_link_nan.py::test_link_nan
also fails for me

**Review by paltusplintus — APPROVED** [review]

**Pull request merged**